File: FizzBuzz/FizzBuzz.hs
main = mapM_ (putStrLn . f) [1..100]

f :: Int -> String
f x
  | g 15 = "FizzBuzz"
  | g 3 = "Fizz"
  | g 5 = "Buzz"
  | otherwise = show x
  where g a = x `mod` a == 0

File: FizzBuzz/FizzBuzz.cpp
#include <iostream>

int main(int argc, char* argv[]) {
	for (int i=1; i<=100; i++) {
		if (!(i % 3)) 
			std::cout << "Fizz";
		if (!(i % 5))
			std::cout << "Buzz";
		else if ((i % 3) && (i % 5))
			std::cout << i;
		std::cout << std::endl;
	}

	return 0;
}
